HW3/IO/Write.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.IO;
+
+namespace HW3.IO
+{
+    public static class Write
+    {
+        public static void AppendCredentials(string filename, string username, string password)
+        {
+            CheckForNewLine(filename);
+
+            using (var fs = new FileStream(filename, FileMode.Append))
+            {
+                using (var sw = new StreamWriter(fs))
+                {
+                    sw.WriteLine($"{username}\t{password}");
+                }
+            }
+        }
+
+        private static void CheckForNewLine(string filename)
+        {
+            string contents;
+
+            using (var fs = new FileStream(filename, FileMode.Open))
+            {
+                var sr = new StreamReader(fs);
+                contents = sr.ReadToEnd();
+
+                if (!contents.EndsWith(Environment.NewLine))
+                {
+                    var sw = new StreamWriter(fs);
+                    sw.Write(Environment.NewLine);
+                    sw.Flush();
```

Comment:
Flush is automatically called when a `StreamWriter` is disposed.  Speaking of dispose...I don't see a using here.  You can "get away" with it when using `FileStream` (since it's really only important that at least one of them be disposed).  Also, if you disposed the `StreamReader` this method would actually fail....hmmm....

I think I would open a `FileStream` twice.  First, open it and read it like you did (except dispose the `StreamReader`).  Then open it in append mode if the newline is needed.  But I think your approach is ok too.